src/pal/src/misc/utils.cpp

```diff
@@ -323,3 +329,72 @@ void UTIL_SetLastErrorFromMach(kern_return_t MachReturn)
 }
 #endif //HAVE_VM_ALLOCATE
 
+#ifdef __APPLE__
+
+#define DARWIN_VERSION_MOJAVE 18
+
+/*++
+Function:
+  GetDarwinVersion() - Helper function to get the version of the Apple's darwin
+--*/
+static UINT GetDarwinVersion()
+{
+    static UINT version = 0;
+
+    if (!version)
+    {
+        char str[32] = {0};
+        size_t size = sizeof(str);
+        int mib[2];
+
+        mib[0] = CTL_KERN;
+        mib[1] = KERN_OSRELEASE;
+        int err = sysctl(mib, 2, str, &size, NULL, 0);
+        _ASSERTE(err == 0);
+        int cnt = sscanf(str, "%d", &version);
+        _ASSERTE(cnt == 1);
+        _ASSERTE(version > 0);
+    }
+
+    return version;
+}
+
+/*++
+Function:
+  IsRunningOnMojaveHardenedRuntime() - Test if the current process is running on Mojave hardened runtime
+--*/
+BOOL IsRunningOnMojaveHardenedRuntime()
+{
+    static int isRunningOnMojaveHardenedRuntime = -1;
+
+    if (isRunningOnMojaveHardenedRuntime == -1)
+    {
+        if (GetDarwinVersion() >= DARWIN_VERSION_MOJAVE)
```

Comment:
Do we need to bother with this version check at all? Given the Apple ecosystem upgrades very fast, this is going to be just an expensive way to compute "true" in like a year.